### ISSUE TITLE
trackEcommProduct produces no collect request 

### ISSUE THREAD

**Issue by thomasdtucker** [rangle/redux-beacon#254]
This is a...
----
 - [x] :beetle: Bug Report
 - [ ] :rocket: Feature Request
 - [ ] :scroll: Documentation Request

Which version of Redux Beacon are you using?
----
 - 2.0.2

Which target(s) are you using?
----
 - [x] Google Analytics
 - [ ] Google Analytics (gtag)
 - [ ] React Native Google Analytics
 - [ ] Google Tag Manager
 - [ ] React Native Google Tag Manager
 - [ ] Amplitude
 - [ ] Segment
 - [ ] Other/Third Party: ...(please specify here)

https://stackoverflow.com/questions/49986398/redux-becon-trackecommproduct-not-firing-a-collect-requestredux-beacon 

your redux-beacon tag does not exist on stackoverflow.

**Comment by ttmarek** [comment]
hey, @thomasdtucker. If possible could you run Google Analytics in debug mode and share what happens when those events fire?

Here are some instructions: https://developers.google.com/analytics/devguides/collection/analyticsjs/debugging

> your redux-beacon tag does not exist on stackoverflow.

:+1: :100: Sorry about that, there should be one now.

**Comment by thomasdtucker** [comment]
@ttmarek Thank you for the great suggestion. 

Here's what I'm seeing. If I include just include the ec plugin, I get this error in the GA debugger: 
`` 
ga('require', 'ec');
``
![image](https://user-images.githubusercontent.com/1490034/39165620-fb9829f8-474a-11e8-94d5-b2723f1e9d38.png)

However, if I incliude the ecommerce plugin as well, I get this error in the GA debugger: 
`` 
ga('require', 'ecommerce');
``
![image](https://user-images.githubusercontent.com/1490034/39165671-3754dc02-474b-11e8-93c0-faa515e67a4e.png)

I have also tried only including the 'ec' plugin, which results in the first error of not having the plugin. 

Any ideas? 

Here is the full stack trace I'm getting 

```
Executing Google Analytics commands.
VM3971 analytics_debug.js:10 Running command: ga("ecommerce:addAction", undefined, {id: "E4700D80AB", affiliation: "Web", revenue: "468.14", tax: undefined, shipping: undefined, coupon: null, list: undefined, step: undefined, option: undefined})
index.js:2178 Plugin "ecommerce" does not have method "addAction".
__stack_frame_overlay_proxy_console__ @ index.js:2178
log @ VM3971 analytics_debug.js:10
O @ VM3971 analytics_debug.js:9
jf.R @ VM3971 analytics_debug.js:79
jf.H @ VM3971 analytics_debug.js:76
Z @ VM3971 analytics_debug.js:79
addAction @ google-analytics.js:36
callEvent @ google-analytics.js:40
(anonymous) @ google-analytics.js:42
(anonymous) @ google-analytics.js:11
passEventsToTarget @ register-events.js:23
handleEvents @ register-events.js:28
registerEvents @ register-events.js:48
(anonymous) @ create-middleware.js:20
(anonymous) @ index.js:14
dispatch @ applyMiddleware.js:35
(anonymous) @ index.js:644
Promise.then (async)
trackBookNow @ index.js:641
(anonymous) @ index.js:214
Promise.then (async)
(anonymous) @ index.js:207
(anonymous) @ index.js:11
(anonymous) @ bindActionCreators.js:3
handleCreateBooking @ index.js:51
onClick @ index.js:86
EnhancedButton._this.handleClick @ EnhancedButton.js:146
callCallback @ react-dom.development.js:542
invokeGuardedCallbackDev @ react-dom.development.js:581
invokeGuardedCallback @ react-dom.development.js:438
invokeGuardedCallbackAndCatchFirstError @ react-dom.development.js:452
executeDispatch @ react-dom.development.js:836
executeDispatchesInOrder @ react-dom.development.js:858
executeDispatchesAndRelease @ react-dom.development.js:956
executeDispatchesAndReleaseTopLevel @ react-dom.development.js:967
forEachAccumulated @ react-dom.development.js:935
processEventQueue @ react-dom.development.js:1112
runEventQueueInBatch @ react-dom.development.js:3607
handleTopLevel @ react-dom.development.js:3616
handleTopLevelImpl @ react-dom.development.js:3347
batchedUpdates @ react-dom.development.js:11082
batchedUpdates @ react-dom.development.js:2330
dispatchEvent @ react-dom.development.js:3421
index.js:2178 Error calling a plugin method: {0: "ecommerce:addAction", 1: undefined, 2: {id: "E4700D80AB", affiliation: "Web", revenue: "468.14", tax: undefined, shipping: undefined, coupon: null, list: undefined, step: undefined, option: undefined}}
__stack_frame_overlay_proxy_console__ @ index.js:2178
log @ VM3971 analytics_debug.js:10
O @ VM3971 analytics_debug.js:9
jf.R @ VM3971 analytics_debug.js:79
jf.H @ VM3971 analytics_debug.js:76
Z @ VM3971 analytics_debug.js:79
addAction @ google-analytics.js:36
callEvent @ google-analytics.js:40
(anonymous) @ google-analytics.js:42
(anonymous) @ google-analytics.js:11
passEventsToTarget @ register-events.js:23
handleEvents @ register-events.js:28
registerEvents @ register-events.js:48
(anonymous) @ create-middleware.js:20
(anonymous) @ index.js:14
dispatch @ applyMiddleware.js:35
(anonymous) @ index.js:644
Promise.then (async)
trackBookNow @ index.js:641
(anonymous) @ index.js:214
Promise.then (async)
(anonymous) @ index.js:207
(anonymous) @ index.js:11
(anonymous) @ bindActionCreators.js:3
handleCreateBooking @ index.js:51
onClick @ index.js:86
EnhancedButton._this.handleClick @ EnhancedButton.js:146
callCallback @ react-dom.development.js:542
invokeGuardedCallbackDev @ react-dom.development.js:581
invokeGuardedCallback @ react-dom.development.js:438
invokeGuardedCallbackAndCatchFirstError @ react-dom.development.js:452
executeDispatch @ react-dom.development.js:836
executeDispatchesInOrder @ react-dom.development.js:858
executeDispatchesAndRelease @ react-dom.development.js:956
executeDispatchesAndReleaseTopLevel @ react-dom.development.js:967
forEachAccumulated @ react-dom.development.js:935
processEventQueue @ react-dom.development.js:1112
runEventQueueInBatch @ react-dom.development.js:3607
handleTopLevel @ react-dom.development.js:3616
handleTopLevelImpl @ react-dom.development.js:3347
batchedUpdates @ react-dom.development.js:11082
batchedUpdates @ react-dom.development.js:2330
dispatchEvent @ react-dom.development.js:3421
VM3971 analytics_debug.js:10 Executing Google Analytics commands.
VM3971 analytics_debug.js:10 Running command: ga("ecommerce:addProduct", {id: "E4700D80AB", name: "Austin", brand: undefined, category: "create-reservation", variant: undefined, price: "468.14", quantity: 1, coupon: null, position: undefined})
index.js:2178 Plugin "ecommerce" does not have method "addProduct".
__stack_frame_overlay_proxy_console__ @ index.js:2178
log @ VM3971 analytics_debug.js:10
O @ VM3971 analytics_debug.js:9
jf.R @ VM3971 analytics_debug.js:79
jf.H @ VM3971 analytics_debug.js:76
Z @ VM3971 analytics_debug.js:79
addProduct @ google-analytics.js:30
callEvent @ google-analytics.js:40
(anonymous) @ google-analytics.js:42
(anonymous) @ google-analytics.js:11
passEventsToTarget @ register-events.js:23
handleEvents @ register-events.js:28
registerEvents @ register-events.js:48
(anonymous) @ create-middleware.js:20
(anonymous) @ index.js:14
dispatch @ applyMiddleware.js:35
(anonymous) @ index.js:645
Promise.then (async)
trackBookNow @ index.js:641
(anonymous) @ index.js:214
Promise.then (async)
(anonymous) @ index.js:207
(anonymous) @ index.js:11
(anonymous) @ bindActionCreators.js:3
handleCreateBooking @ index.js:51
onClick @ index.js:86
EnhancedButton._this.handleClick @ EnhancedButton.js:146
callCallback @ react-dom.development.js:542
invokeGuardedCallbackDev @ react-dom.development.js:581
invokeGuardedCallback @ react-dom.development.js:438
invokeGuardedCallbackAndCatchFirstError @ react-dom.development.js:452
executeDispatch @ react-dom.development.js:836
executeDispatchesInOrder @ react-dom.development.js:858
executeDispatchesAndRelease @ react-dom.development.js:956
executeDispatchesAndReleaseTopLevel @ react-dom.development.js:967
forEachAccumulated @ react-dom.development.js:935
processEventQueue @ react-dom.development.js:1112
runEventQueueInBatch @ react-dom.development.js:3607
handleTopLevel @ react-dom.development.js:3616
handleTopLevelImpl @ react-dom.development.js:3347
batchedUpdates @ react-dom.development.js:11082
batchedUpdates @ react-dom.development.js:2330
dispatchEvent @ react-dom.development.js:3421
index.js:2178 Error calling a plugin method: {0: "ecommerce:addProduct", 1: {id: "E4700D80AB", name: "Austin", brand: undefined, category: "create-reservation", variant: undefined, price: "468.14", quantity: 1, coupon: null, position: undefined}}
__stack_frame_overlay_proxy_console__ @ index.js:2178
log @ VM3971 analytics_debug.js:10
O @ VM3971 analytics_debug.js:9
jf.R @ VM3971 analytics_debug.js:79
jf.H @ VM3971 analytics_debug.js:76
Z @ VM3971 analytics_debug.js:79
addProduct @ google-analytics.js:30
callEvent @ google-analytics.js:40
(anonymous) @ google-analytics.js:42
(anonymous) @ google-analytics.js:11
passEventsToTarget @ register-events.js:23
handleEvents @ register-events.js:28
registerEvents @ register-events.js:48
(anonymous) @ create-middleware.js:20
(anonymous) @ index.js:14
dispatch @ applyMiddleware.js:35
(anonymous) @ index.js:645
Promise.then (async)
trackBookNow @ index.js:641
(anonymous) @ index.js:214
Promise.then (async)
(anonymous) @ index.js:207
(anonymous) @ index.js:11
(anonymous) @ bindActionCreators.js:3
handleCreateBooking @ index.js:51
onClick @ index.js:86
EnhancedButton._this.handleClick @ EnhancedButton.js:146
callCallback @ react-dom.development.js:542
invokeGuardedCallbackDev @ react-dom.development.js:581
invokeGuardedCallback @ react-dom.development.js:438
invokeGuardedCallbackAndCatchFirstError @ react-dom.development.js:452
executeDispatch @ react-dom.development.js:836
executeDispatchesInOrder @ react-dom.development.js:858
executeDispatchesAndRelease @ react-dom.development.js:956
executeDispatchesAndReleaseTopLevel @ react-dom.development.js:967
forEachAccumulated @ react-dom.development.js:935
processEventQueue @ react-dom.development.js:1112
runEventQueueInBatch @ react-dom.development.js:3607
handleTopLevel @ react-dom.development.js:3616
handleTopLevelImpl @ react-dom.development.js:3347
batchedUpdates @ react-dom.development.js:11082
batchedUpdates @ react-dom.development.js:2330
dispatchEvent @ react-dom.development.js:3421

```

**Comment by thomasdtucker** [comment]
It's sending ecommerce:addProduct instead of ec:addProduct. Is there something I missed in my config?

**Comment by ttmarek** [comment]
Nope, this looks to be an issue with the event helpers. I'm just having dinner right now, but I should be able to push a patch up in the next few hours. I'll let you know once the fix is available...shouldn't be too long now.

**Comment by ttmarek** [comment]
@thomasdtucker the patch is out. Please upgrade to v.1.0.1 of the target:

```
npm install @redux-beacon/google-analytics@1.0.1
```

Let me know if everything's sorted!

**Comment by thomasdtucker** [comment]
![image](https://user-images.githubusercontent.com/1490034/39322054-3f0d6a4a-494e-11e8-89d5-933a3e2d26fd.png)
![image](https://user-images.githubusercontent.com/1490034/39322067-48ba484c-494e-11e8-9615-863e0689b5f2.png)
![image](https://user-images.githubusercontent.com/1490034/39322086-5685c24e-494e-11e8-885c-061559ef6dba.png)
![image](https://user-images.githubusercontent.com/1490034/39322097-5e96d9aa-494e-11e8-8e3d-df9040d589a6.png)

**Comment by thomasdtucker** [comment]
@ttmarek any ideas here?

**Comment by ttmarek** [comment]
hey @thomasdtucker, sorry I've had quite a busy week. I'll take a look in a few hours.

**Comment by thomasdtucker** [comment]
Not at all! I’m so grateful for any time you can get to it! I’m also
forking and trying to figure it out on my own so I can issues a PR. :)

Great work, man.
- Thomas

**Comment by ttmarek** [comment]
I'm not sure whether the Google Analytics API changed from `ec:addAction` to `ec:setAction`, or if it was a mistake on our end. Regardless, it is definitely `ec:setAction` now and I've pushed yet another update that should fix that error for you. `npm install @redux-beacon/google-analytics@1.0.2`.

I'm not sure I understand the second issue you were describing, it looks like you have four actions, and ga is being called four time.

**Comment by thomasdtucker** [comment]
@ttmarek thank you for the update. 

I downloaded the new version and now neither enhanced call is firing. 

![image](https://user-images.githubusercontent.com/1490034/39481275-81c34d24-4d30-11e8-82d3-c2ac2b4bb698.png)

![image](https://user-images.githubusercontent.com/1490034/39487078-ae221a72-4d43-11e8-9164-2177436f7ff4.png)

**Comment by ttmarek** [comment]
Looks like you're missing the `actionName` when creating the ecommAction: https://rangle.gitbook.io/redux-beacon/targets/googleanalytics#ecommaction.

As far as I can tell that's the only thing that looks "off" about the calls that Redux Beacon is making fo you. Just out of curiosity, if you open up a console and make those calls yourself do they come through?

**Comment by thomasdtucker** [comment]
Code fixed in 1.0.2 :)
